userspace/libsinsp/parsers.cpp

```diff
@@ -2495,15 +2499,160 @@ void sinsp_parser::parse_bind_exit(sinsp_evt *evt)
 	}
 }
 
+/**
+ * Register a socket in pending state
+ */
+void sinsp_parser::parse_connect_enter(sinsp_evt *evt){
+    if (!m_track_connection_pending){
+        return;
+    }
+
+    sinsp_evt_param *parinfo;
+    uint8_t *packed_data;
+
+    if(evt->m_fdinfo == NULL)
+    {
+        return;
+    }
+
+    if(m_track_connection_status)
+    {
+        evt->m_fdinfo->set_socket_pending();
+    }
+
+    parinfo = evt->get_param(1);
+    if(parinfo->m_len == 0)
+    {
+        //
+        // No address, there's nothing we can really do with this.
+        // This happens for socket types that we don't support, so we have the assertion
+        // to make sure that this is not a type of socket that we support.
+        //
+        ASSERT(!(evt->m_fdinfo->is_unix_socket() || evt->m_fdinfo->is_ipv4_socket()));
+        return;
+    }
+
+    packed_data = (uint8_t*)parinfo->m_val;
+
+    fill_socket_info(evt, packed_data);
+
+    //
+    // If there's a listener callback, invoke it
+    //
+    if(m_fd_listener)
+    {
+        m_fd_listener->on_connect(evt, packed_data);
+    }
```

Comment:
AIUI, this is the real point of the PR, right?

Reply:
Yes.